<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>CameraWidget</class>
    <widget class="QWidget" name="CameraWidget">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>960</width>
                <height>640</height>
            </rect>
        </property>
        <property name="windowTitle">
            <string>Camera Viewer</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout" stretch="9,1">
            <property name="spacing">
                <number>0</number>
            </property>
            <property name="leftMargin">
                <number>0</number>
            </property>
            <property name="topMargin">
                <number>0</number>
            </property>
            <property name="rightMargin">
                <number>0</number>
            </property>
            <property name="bottomMargin">
                <number>0</number>
            </property>
            <item>
                <layout class="QVBoxLayout" name="view_layout">
                    <property name="spacing">
                        <number>0</number>
                    </property>
                    <item>
                        <widget class="QLabel" name="camera_label">
                            <property name="text">
                                <string>instant display</string>
                            </property>
                            <property name="alignment">
                                <set>Qt::AlignCenter</set>
                            </property>
                        </widget>
                    </item>
                    <item>
                        <widget class="QStackedWidget" name="stacked_widget">
                            <property name="sizePolicy">
                                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                                    <horstretch>0</horstretch>
                                    <verstretch>0</verstretch>
                                </sizepolicy>
                            </property>
                            <property name="palette">
                                <palette>
                                    <active>
                                        <colorrole role="Base">
                                            <brush brushstyle="SolidPattern">
                                                <color alpha="255">
                                                    <red>255</red>
                                                    <green>255</green>
                                                    <blue>255</blue>
                                                </color>
                                            </brush>
                                        </colorrole>
                                        <colorrole role="Window">
                                            <brush brushstyle="SolidPattern">
                                                <color alpha="255">
                                                    <red>145</red>
                                                    <green>145</green>
                                                    <blue>145</blue>
                                                </color>
                                            </brush>
                                        </colorrole>
                                    </active>
                                    <inactive>
                                        <colorrole role="Base">
                                            <brush brushstyle="SolidPattern">
                                                <color alpha="255">
                                                    <red>255</red>
                                                    <green>255</green>
                                                    <blue>255</blue>
                                                </color>
                                            </brush>
                                        </colorrole>
                                        <colorrole role="Window">
                                            <brush brushstyle="SolidPattern">
                                                <color alpha="255">
                                                    <red>145</red>
                                                    <green>145</green>
                                                    <blue>145</blue>
                                                </color>
                                            </brush>
                                        </colorrole>
                                    </inactive>
                                    <disabled>
                                        <colorrole role="Base">
                                            <brush brushstyle="SolidPattern">
                                                <color alpha="255">
                                                    <red>145</red>
                                                    <green>145</green>
                                                    <blue>145</blue>
                                                </color>
                                            </brush>
                                        </colorrole>
                                        <colorrole role="Window">
                                            <brush brushstyle="SolidPattern">
                                                <color alpha="255">
                                                    <red>145</red>
                                                    <green>145</green>
                                                    <blue>145</blue>
                                                </color>
                                            </brush>
                                        </colorrole>
                                    </disabled>
                                </palette>
                            </property>
                            <property name="lineWidth">
                                <number>0</number>
                            </property>
                            <property name="currentIndex">
                                <number>0</number>
                            </property>
                            <widget class="QWidget" name="viewfinder_page">
                                <layout class="QGridLayout" name="gridLayout_5">
                                    <property name="leftMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="topMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="rightMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="bottomMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="spacing">
                                        <number>0</number>
                                    </property>
                                    <item row="0" column="0">
                                        <widget class="QCameraViewfinder" name="viewfinder" native="true">
                                            <property name="styleSheet">
                                                <string notr="true"/>
                                            </property>
                                        </widget>
                                    </item>
                                </layout>
                            </widget>
                            <widget class="QWidget" name="preview_page">
                                <layout class="QGridLayout" name="gridLayout_4">
                                    <property name="leftMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="topMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="rightMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="bottomMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="spacing">
                                        <number>0</number>
                                    </property>
                                    <item row="0" column="0">
                                        <widget class="QLabel" name="preview_label">
                                            <property name="sizePolicy">
                                                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                                                    <horstretch>0</horstretch>
                                                    <verstretch>0</verstretch>
                                                </sizepolicy>
                                            </property>
                                            <property name="frameShape">
                                                <enum>QFrame::Box</enum>
                                            </property>
                                            <property name="lineWidth">
                                                <number>0</number>
                                            </property>
                                            <property name="text">
                                                <string/>
                                            </property>
                                            <property name="alignment">
                                                <set>Qt::AlignCenter</set>
                                            </property>
                                        </widget>
                                    </item>
                                </layout>
                            </widget>
                        </widget>
                    </item>
                </layout>
            </item>
            <item>
                <layout class="QVBoxLayout" name="ctrl_layout" stretch="1,20">
                    <property name="spacing">
                        <number>0</number>
                    </property>
                    <property name="leftMargin">
                        <number>0</number>
                    </property>
                    <property name="topMargin">
                        <number>0</number>
                    </property>
                    <property name="rightMargin">
                        <number>0</number>
                    </property>
                    <property name="bottomMargin">
                        <number>0</number>
                    </property>
                    <item>
                        <widget class="QToolButton" name="lock_btn">
                            <property name="sizePolicy">
                                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                                    <horstretch>0</horstretch>
                                    <verstretch>0</verstretch>
                                </sizepolicy>
                            </property>
                            <property name="styleSheet">
                                <string notr="true">background-color: rgb(114, 159, 207);</string>
                            </property>
                            <property name="text">
                                <string>Focus</string>
                            </property>
                        </widget>
                    </item>
                    <item>
                        <widget class="QTabWidget" name="ctrl_tab_widget">
                            <property name="currentIndex">
                                <number>0</number>
                            </property>
                            <widget class="QWidget" name="tab_image">
                                <attribute name="title">
                                    <string>Image</string>
                                </attribute>
                                <layout class="QGridLayout" name="gridLayout">
                                    <property name="leftMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="topMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="rightMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="bottomMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="spacing">
                                        <number>0</number>
                                    </property>
                                    <item row="2" column="0">
                                        <widget class="QLabel" name="exposure_label">
                                            <property name="text">
                                                <string>Exposure Compensation:</string>
                                            </property>
                                        </widget>
                                    </item>
                                    <item row="1" column="0">
                                        <spacer name="exposure_spacer">
                                            <property name="orientation">
                                                <enum>Qt::Vertical</enum>
                                            </property>
                                            <property name="sizeHint" stdset="0">
                                                <size>
                                                    <width>20</width>
                                                    <height>161</height>
                                                </size>
                                            </property>
                                        </spacer>
                                    </item>
                                    <item row="3" column="0">
                                        <widget class="QSlider" name="exposure_slider">
                                            <property name="sizePolicy">
                                                <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                                                    <horstretch>0</horstretch>
                                                    <verstretch>0</verstretch>
                                                </sizepolicy>
                                            </property>
                                            <property name="styleSheet">
                                                <string notr="true">QSlider::handle:horizontal {
                                                    background: QRadialGradient(cx:0, cy:0, radius: 1, fx:0.5, fy:0.5,
                                                    stop:0 white, stop:1 green);
                                                    width: 16px;
                                                    height: 16px;
                                                    margin: -5px 6px -5px 6px;
                                                    border-radius:11px;
                                                    border: 3px solid black;
                                                    }
                                                    QSlider::groove:horizontal {
                                                    height: 12px;
                                                    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #B1B1B1,
                                                    stop:1 #c4c4c4);
                                                    margin: 2px 0
                                                    }
                                                    QSlider {
                                                    background-color: white;
                                                    border-style: outset;
                                                    border-radius: 10px;
                                                    }
                                                </string>
                                            </property>
                                            <property name="minimum">
                                                <number>-4</number>
                                            </property>
                                            <property name="maximum">
                                                <number>4</number>
                                            </property>
                                            <property name="pageStep">
                                                <number>2</number>
                                            </property>
                                            <property name="orientation">
                                                <enum>Qt::Horizontal</enum>
                                            </property>
                                            <property name="tickPosition">
                                                <enum>QSlider::TicksAbove</enum>
                                            </property>
                                        </widget>
                                    </item>
                                    <item row="0" column="0">
                                        <widget class="QToolButton" name="shot_btn">
                                            <property name="enabled">
                                                <bool>false</bool>
                                            </property>
                                            <property name="sizePolicy">
                                                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                                                    <horstretch>0</horstretch>
                                                    <verstretch>0</verstretch>
                                                </sizepolicy>
                                            </property>
                                            <property name="styleSheet">
                                                <string notr="true">background-color: rgb(92, 53, 102);
                                                    color: rgb(255, 255, 255);
                                                </string>
                                            </property>
                                            <property name="text">
                                                <string>Capture Photo</string>
                                            </property>
                                            <property name="icon">
                                                <iconset resource="../../resources/app.qrc">
                                                    <normaloff>:/img/shutter.png</normaloff>:/img/shutter.png
                                                </iconset>
                                            </property>
                                            <property name="iconSize">
                                                <size>
                                                    <width>32</width>
                                                    <height>32</height>
                                                </size>
                                            </property>
                                            <property name="toolButtonStyle">
                                                <enum>Qt::ToolButtonTextBesideIcon</enum>
                                            </property>
                                        </widget>
                                    </item>
                                </layout>
                            </widget>
                            <widget class="QWidget" name="tab_device">
                                <attribute name="title">
                                    <string>Device</string>
                                </attribute>
                                <layout class="QVBoxLayout" name="verticalLayout">
                                    <property name="spacing">
                                        <number>0</number>
                                    </property>
                                    <property name="leftMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="topMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="rightMargin">
                                        <number>0</number>
                                    </property>
                                    <property name="bottomMargin">
                                        <number>0</number>
                                    </property>
                                    <item>
                                        <widget class="QListWidget" name="device_list">
                                            <property name="sizePolicy">
                                                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                                                    <horstretch>0</horstretch>
                                                    <verstretch>0</verstretch>
                                                </sizepolicy>
                                            </property>
                                            <property name="editTriggers">
                                                <set>
                                                    QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed
                                                </set>
                                            </property>
                                            <property name="showDropIndicator" stdset="0">
                                                <bool>false</bool>
                                            </property>
                                            <property name="selectionMode">
                                                <enum>QAbstractItemView::SingleSelection</enum>
                                            </property>
                                            <property name="selectionBehavior">
                                                <enum>QAbstractItemView::SelectItems</enum>
                                            </property>
                                            <property name="textElideMode">
                                                <enum>Qt::ElideNone</enum>
                                            </property>
                                            <property name="resizeMode">
                                                <enum>QListView::Adjust</enum>
                                            </property>
                                            <property name="wordWrap">
                                                <bool>true</bool>
                                            </property>
                                            <property name="selectionRectVisible">
                                                <bool>false</bool>
                                            </property>
                                            <property name="itemAlignment">
                                                <set>Qt::AlignLeading</set>
                                            </property>
                                            <property name="sortingEnabled">
                                                <bool>true</bool>
                                            </property>
                                        </widget>
                                    </item>
                                </layout>
                            </widget>
                        </widget>
                    </item>
                </layout>
            </item>
        </layout>
    </widget>
    <customwidgets>
        <customwidget>
            <class>QCameraViewfinder</class>
            <extends>QWidget</extends>
            <header>qcameraviewfinder.h</header>
            <container>1</container>
        </customwidget>
    </customwidgets>
    <resources>
        <include location="../../resources/app.qrc"/>
    </resources>
    <connections/>
</ui>
